daemon/cmd/daemon.go

```diff
@@ -484,18 +484,10 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	}
 	bootstrapStats.restore.End(true)
 
+	// Load cached information from restored endpoints in to FQDN NameManager and DNS proxies
 	bootstrapStats.fqdn.Start()
-	err = params.DNSProxy.BootstrapFQDN(restoredEndpoints.possible, option.Config.ToFQDNsPreCache)
-	if err != nil {
-		bootstrapStats.fqdn.EndError(err)
-		return nil, restoredEndpoints, err
-	}
-
-	// This is done in preCleanup so that proxy stops serving DNS traffic before shutdown
-	cleaner.preCleanupFuncs.Add(func() {
```

Comment:
note to reviewers: there's no obvious need for ordering between dns proxy shutdown and daemon shutdown.

Reply:
It's hard for me to understand what the implications of daemon shutdown are. As in, do endpoint lookups fail now? Can there be a race where the DNS proxy receives a request it fails to handle because it can't lookup the source EP?

Reply:
I suppose there could be the odd request that gets a SERVFAIL instead of timing out, but the effect is the same: we are immediately closing the socket and dropping all in-flight requests. We've never made any effort to gracefully shut down -- which would be nice -- but carefully ordering exactly *how* we drop requests on the floor is not particularly interesting.

Reply:
Plan: file a follow-up PR that moves the DNSPolicyUnloadOnShutdown in to the fqdn proxy cell and out of the daemon. Then we have full control of the lifecycle.

Reply:
Drive-by thought: Could we get rid of DNSPolicyUnloadOnShutdown? I'm not sure if we have any users that actually enable it. Worth checking with the community